spring-cloud-gcp-autoconfigure/src/main/java/org/springframework/cloud/gcp/autoconfigure/metrics/GcpStackdriverMetricsAutoConfiguration.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright 2017-2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.gcp.autoconfigure.metrics;
+
+import java.io.IOException;
+
+import com.google.api.gax.core.CredentialsProvider;
+import io.micrometer.core.instrument.Clock;
+import io.micrometer.core.instrument.step.StepMeterRegistry;
+import io.micrometer.stackdriver.StackdriverConfig;
+
+import org.springframework.boot.actuate.autoconfigure.metrics.export.stackdriver.StackdriverMetricsExportAutoConfiguration;
+import org.springframework.boot.actuate.autoconfigure.metrics.export.stackdriver.StackdriverProperties;
+import org.springframework.boot.autoconfigure.AutoConfigureBefore;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnBean;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnClass;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.cloud.gcp.core.DefaultCredentialsProvider;
+import org.springframework.cloud.gcp.core.GcpProjectIdProvider;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+
+/**
+ * Provides auto-detection for project.
+ *
+ * @author Eddú Meléndez
+ */
+@Configuration
+@AutoConfigureBefore(StackdriverMetricsExportAutoConfiguration.class)
+@ConditionalOnClass({StepMeterRegistry.class, StackdriverConfig.class})
+@ConditionalOnBean(Clock.class)
```

Comment:
Don't think so. Since all of them are provided for `spring-boot` once `actuator`, `micrometer` and `micrometer-stackdriver` are declared. When the conditions meet we this auto-config triggers.